Docs/docker-compose guide

bcatcho opened the pull request:
I use docker-compose for my local development setup because minikube is too resource intensive for my team. Ambassador on top of docker-compose is a fast way to experiment with different features really quickly. While there are differences between how a kuberenetes and docker-compose environment operate all Ambassador features work in both.

The reason I am opening this PR is two fold:
1. To make it easy for more people to discover how useful Ambassador is
2. To continue the docker-compose support (though not for production use)

bcatcho commented:
It seems like the travis-ci check has been running into issues for a couple of days:
```
openssl aes-256-cbc \
    -K $encrypted_95ce0d75e6f7_key \
    -iv $encrypted_95ce0d75e6f7_iv \
    -in kubernaut-config.json.enc \
    -out ~/.config/kubernaut/config.json \
    -d
iv undefined
```

richarddli commented:
Thanks! @nbkrause can you try out this tutorial please?

nbkrause commented:
> It seems like the travis-ci check has been running into issues for a couple of days:
> ```
> openssl aes-256-cbc \
>     -K $encrypted_95ce0d75e6f7_key \
>     -iv $encrypted_95ce0d75e6f7_iv \
>     -in kubernaut-config.json.enc \
>     -out ~/.config/kubernaut/config.json \
>     -d
>  iv undefined
> ```
This is due to a secret that is not public.

nbkrause commented on docs/user-guide/docker-compose.md:
```diff
@@ -0,0 +1,380 @@
+    image: quay.io/datawire/ambassador:0.38.0
+    ports:
+    # expose port 80
+    - 80
```
Change to 
```
ports:
- target: 80
  published: 8080
```

nbkrause commented on docs/user-guide/docker-compose.md:
```diff
@@ -0,0 +1,380 @@
+    image: quay.io/datawire/ambassador:0.38.0
+    ports:
+    # expose port 80
+    - 80
```
Change to 
```
ports:
- target: 80
  published: 8080
```

nbkrause commented on docs/user-guide/docker-compose.md:
```diff
@@ -0,0 +1,380 @@
+    image: quay.io/datawire/ambassador:0.38.0
+    ports:
+    # expose port 80
+    - 80
```
Change to 
```
ports:
- target: 80
  published: 8080
```

nbkrause commented on docs/user-guide/docker-compose.md:
```diff
@@ -0,0 +1,380 @@
+docker-compose up -d -V ambassador
+
+# curl the quote-of-the-moment service
+curl localhost:8080/qotm/quote
```
`curl localhost:8080/qotm/quote` is not accessible. Change to either `curl localhost:8080/qotm/quote/1` for select quote or `curl localhost:8080/qotm/` for random quote.

nbkrause commented on docs/user-guide/docker-compose.md:
```diff
@@ -0,0 +1,380 @@
+docker-compose up -d -V ambassador
+
+# curl the quote-of-the-moment service
+curl localhost:8080/qotm/quote
```
`curl localhost:8080/qotm/quote` is not accessible. Change to either `curl localhost:8080/qotm/quote/1` for select quote or `curl localhost:8080/qotm/` for random quote.

nbkrause commented on docs/user-guide/docker-compose.md:
```diff
@@ -0,0 +1,380 @@
+docker-compose up -d -V ambassador
+
+# curl the quote-of-the-moment service as many times as you would like
+curl -I localhost:8080/qotm/quote
```
`curl localhost:8080/qotm/quote` is not accessible. Change to either `curl localhost:8080/qotm/quote/1` for select quote or `curl localhost:8080/qotm/` for random quote.

bcatcho commented:
@nbkrause I have addressed your comments in the latest commit though I used the shorthand version for port mapping eg `8080:80`. I can use the longhand version if you desire.
